src/main/java/org/wso2/carbon/connector/connection/MailBoxConnection.java

```diff
@@ -0,0 +1,142 @@
+        } catch (MessagingException e) {
+            throw new EmailConnectionException(format("Error occurred while connecting to the store. %s",
+                    e.getMessage()), e);
+        }
+    }
+
+    /**
+     * Opens and return the email folder.
+     * <p>
+     * If there was an already opened folder and a different one is requested the opened folder will be closed
+     * and the new one will be opened.
+     *
+     * @param mailBoxFolder the name of the folder to be opened.
+     * @param openMode      open the folder in READ_ONLY or READ_WRITE mode
+     * @return the opened Folder
+     */
+    public synchronized Folder getFolder(String mailBoxFolder, int openMode) throws EmailConnectionException {
+
+        try {
+            if (folder != null) {
+                if (isCurrentFolder(mailBoxFolder) && folder.isOpen() && folder.getMode() == openMode) {
+                    return folder;
+                }
+                closeFolder(false);
+            }
+
+            folder = store.getFolder(mailBoxFolder);
+            folder.open(openMode);
+
+        } catch (MessagingException e) {
+            throw new EmailConnectionException(format("Error while opening folder : %s. %s", mailBoxFolder,
+                    e.getMessage()), e);
+        }
+        return folder;
+    }
+
+    /**
+     * Closes the current connection folder.
+     *
+     * @param expunge whether to remove all the emails marked as DELETED.
+     */
+    public synchronized void closeFolder(boolean expunge) throws EmailConnectionException {
+
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug(format("Closing folder: %s ...", this.folder.getFullName()));
+            }
+            if (folder != null && folder.isOpen()) {
+                folder.close(expunge);
+            }
+        } catch (MessagingException e) {
+            throw new EmailConnectionException(format("Error occurred while closing folder: %s. %s",
+                    this.folder.getFullName(), e.getMessage()), e);
+        }
+    }
+
+    /**
+     * Checks if a mailBoxFolder name is the same name as the current folder.
+     *
+     * @param mailBoxFolder the name of the folder
+     * @return true if is the same folder, false otherwise.
+     */
+    private boolean isCurrentFolder(String mailBoxFolder) {
+
+        return folder.getName() != null && folder.getName().equalsIgnoreCase(mailBoxFolder);
+    }
+
+    /**
+     * Closes the folder and the store
+     */
+    synchronized void disconnect() {
+
+        try {
+            closeFolder(false);
+        } catch (Exception e) {
+            log.error(format("Error closing mailbox folder %s when disconnecting. %s", folder.getName(),
+                    e.getMessage()), e);
+        } finally {
+            try {
+                store.close();
+            } catch (Exception e) {
+                log.error(format("Error closing store when disconnecting. %s", e.getMessage()), e);
```

Comment:
No need of e.getMessage(). It will be captured by logging exception.

Reply:
Fixed